### PR TITLE
#121: Use assertJ everywhere

### PR DESCRIPTION
**Pull request by vpondala**
Migrated all assertions to AssertJ.

### CODE REVIEW

**Comment by vpondala** [comment]
My local build succeeded. This is my 1st Android build, not sure how to fix it, Can you please guide?

**Comment by StefMa** [comment]
Hey, thank you for your PR!

Sure.
Try googeling for "Duplicate files copied in APK META-INF/License".

The problem is that two or more deoencies try to add these file into the
apk. Which is not allowed.
I am not  hundred percent sure but it seems that the assertJ and the
support test runner wants to include that file.

Just try to exclude it and the thirtyinch module  and it should work 👍

**Comment by vpondala** [comment]
I added packagingOptions { exclude 'META-INF/LICENSE' } and the build passes now. Please have a look at the diffs and let me know if the changes look good.

File: thirtyinch/src/test/java/net/grandcentrix/thirtyinch/SendToViewTest.java [hunk 1]
```diff
@@ -118,9 +116,10 @@ public Thread newThread(final Runnable r) {
             public void call(final TiView tiView) {
                 // Then the work gets executed on the ui thread
                 final Thread currentThread = Thread.currentThread();
-                assertNotSame(testThread, currentThread);
-                assertTrue("executed on wrong thread",
-                        "test ui thread".equals(currentThread.getName()));
+                assertThat(testThread).isNotSameAs(currentThread);
+                assertThat("test ui thread")
+                        .as("executed on wrong thread")
```
Comment (passsy) [inline thirtyinch/src/test/java/net/grandcentrix/thirtyinch/SendToViewTest.java]: didn't know this one, great 🎉

File: thirtyinch/src/test/java/net/grandcentrix/thirtyinch/TiLifecycleObserverTest.java [hunk 2]
```diff
@@ -303,14 +302,14 @@ public void onChange(final TiPresenter.State state,
         mPresenter.detachView();
 
         final Object[] beforeLast = states.get(states.size() - 2);
-        assertEquals(beforeLast[0], TiPresenter.State.VIEW_DETACHED);
-        assertEquals(beforeLast[1], false);
-        assertNotNull(beforeLast[2]);
+        assertThat(beforeLast[0]).isEqualTo(TiPresenter.State.VIEW_DETACHED);
+        assertThat(beforeLast[1]).isEqualTo(false);
```
Comment (passsy) [inline thirtyinch/src/test/java/net/grandcentrix/thirtyinch/TiLifecycleObserverTest.java]: better use `.isFalse()`

**Comment by passsy** [comment]
There's a mixup between `import static org.assertj.core.api.Assertions.*;` and `import static org.assertj.core.api.Java6Assertions.*;`. The `Java6Assertions` was required for androidTests in the past. I'm not sure if this is still the case, AssertJ improved Android support a lot recently.

Try to use `import static org.assertj.core.api.Assertions.*;` everywhere

File: thirtyinch/src/test/java/net/grandcentrix/thirtyinch/TiLifecycleObserverTest.java (hunk 2 above)
Comment (vpondala) [reply thirtyinch/src/test/java/net/grandcentrix/thirtyinch/TiLifecycleObserverTest.java]: In these cases, it's AbstractObjectAssert, which does not have isTrue()/isFalse().

**Comment by passsy** [comment]
Thanks for your contribution!

**Pull request merged**